daemon/peer.c

```diff
@@ -911,11 +911,18 @@ static void their_htlc_added(struct peer *peer, struct htlc *htlc,
 
 	//FIXME: dirty trick to retrieve unexported state
 	memcpy(&pk, peer->dstate->secret, sizeof(pk));
+
 	packet = parse_onionpacket(peer,
 				   htlc->routing, tal_count(htlc->routing));
-	if (packet)
-		step = process_onionpacket(packet, packet, &pk, htlc->rhash.u.u8,
-					   sizeof(htlc->rhash));
+	if (packet) {
+		u8 shared_secret[32];
+
+		if (onion_shared_secret(shared_secret, packet, &pk))
```

Comment:
This is the old code, just a change because I split the onion code routines into two parts.